cachi2/core/models/sbom.py

```diff
@@ -93,3 +94,134 @@ class Sbom(pydantic.BaseModel):
     def _unique_components(cls, components: list[Component]) -> list[Component]:
         """Sort and de-duplicate components."""
         return unique_sorted(components, by=lambda component: component.key())
+
+    def to_spdx(self) -> "SPDXSbom":
+        """Convert a CycloneDX SBOM to an SPDX SBOM."""
+        return SPDXSbom(
+            packages=[
+                SPDXPackage(
+                    name=component.name,
+                    version=component.version,
+                    package_download_location=component.purl,
+                )
+                for component in self.components
+            ],
+            creationInfo=SPDXCreationInfo(
+                creators=[f"{tool.vendor} {tool.name}" for tool in self.metadata.tools]
+            ),
+        )
+
+
+class SPDXPackageExternalRef(pydantic.BaseModel):
+    """SPDX Package External Reference.
+
+    Compliant to the SPDX specification:
+    https://spdx.github.io/spdx-spec/v2.3/package-information/#721-external-reference-field
+    """
+
+    referenceCategory: str = pydantic.Field(
+        pattern="^SECURITY|PACKAGE-MANAGER|PERSISTENT-ID|OTHER$"
+    )
+    referenceLocator: str
+    referenceType: str
+
+    @pydantic.model_validator(mode="after")
+    def _model_validator(self: Self) -> Self:
+        if self.referenceCategory == "SECURITY":
+            if self.referenceType not in ["cpe22", "cpe23", "advisory", "fix", "url", "swid"]:
+                raise ValueError("Invalid reference type for SECURITY category")
+        elif self.referenceCategory == "PACKAGE-MANAGER":
+            if self.referenceType not in ["maven-central", "npm", "nuget", "bower", "purl"]:
+                raise ValueError("Invalid reference type for PACKAGE-MANAGER category")
+        elif self.referenceCategory == "PERSISTENT-ID":
+            if self.referenceType not in ["swh", "gitbom"]:
+                raise ValueError("Invalid reference type for PERSISTENT-ID category")
+        return self
+
+    def __hash__(self) -> int:
+        return hash((self.referenceLocator, self.referenceType, self.referenceCategory))
+
+
+class SPDXPackage(pydantic.BaseModel):
+    """SPDX Package.
+
+    Compliant to the SPDX specification:
+    https://spdx.github.io/spdx-spec/v2.3/package-information/
+    """
+
+    name: str
+    version: Optional[str] = None
```

Comment:
Shouldn't it be named `versionInfo`?
https://github.com/RedHatProductSecurity/security-data-guidelines/blob/71fb1006e269b59208cd6206a6d4c2cd7a30b68c/sbom/examples/container_image/build/kernel-module-management-operator-container-1.1.2-25.spdx.json#L17

Reply:
yeah, another docs inconsistency